### PR TITLE
v1.0.0

### PR DESCRIPTION
**Pull request by SokratisVidros**
This is an attempt to bring riviere to 2019 along with latest dependencies and examples using Koa 2. 

Please note that this PR contains breaking changes for riviere middleware builder and error callback. Thus, we've also bumped it's major version to 1.x.x.

Last but not least, unit tests were slightly cleaned.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,14 +97,14 @@ app.listen(3000);
...
-const Riviere = require('riviere');
+const riviere = require('riviere');
 
 // this is just an example, you can use any http library
 const rp = require('request-promise');
 
 const app = new Koa();
 
-app.use(Riviere.middleware());
+app.use(riviere());
```
Comment (nikostoulas) [inline README.md:107]: 👍 nice

File: README.md [hunk 2]
```diff
@@ -128,22 +128,22 @@ app.listen(3000);
...
-const riviereConfObj = {
+const configuration = {
     context: ctx => {
         return {};
     },
     errors: {
         enabled: true,
-        callback: (ctx, error) => {
-            throw(error);
+        callback: (err, ctx) => {
```
Comment (nikostoulas) [inline README.md:145]: To further simplify riviere in the future I would suggest removing error handling entirely. Riviere could catch errors in order to log them and rethrow them as it now does by default.

**Review by nikostoulas — APPROVED** [review]
💯

File: README.md (hunk 2 above)
Comment (SokratisVidros) [reply README.md:145]: 👍

**Pull request merged**